src/solve.c

```diff
@@ -116,42 +118,64 @@ void run(uint8_t speed)
 	char movement;
 	float distance = 0;
 
-	make_smooth_path(run_sequence, smooth_path);
+	make_smooth_path(run_sequence, smooth_path, PATH_DIAGONALS);
 	while (true) {
 		movement = smooth_path[i++];
 		if (movement == MOVE_END)
```

Comment:
Why MOVE_END is not inside the switch case as the rest of movements?

Reply:
We want to break from the `while(true)`. If I move it inside the switch, then break will only break from the switch, right?

Reply:
Then I found that there is no code after the `while` loop, so we can just return...

7c2e4ab74f9a3741ac19d07bcdeb24f07abba632

Added a new commit instead of a fixup because that code is already in master, not anything new added with this "diagonals" implementation. :wink:

Reply:
great